### PR TITLE
docs(refactoring) - laboratory, feature flags

### PR DESCRIPTION
**Pull request by hazelw**
Small doc on safer refactoring

### CODE REVIEW

File: languages/Python/refactoring.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Refactoring
+
+## Safer Refactoring
+
+As with any form of software engineering, refactoring comes with its own set of risks. This section of our style guide focusses on strategies we use at Zego to reduce that risk.
+
+### Experiments
+
+When refactoring existing code with complex behaviour, consider using the [laboratory library](https://github.com/joealcorn/laboratory) to ensure that the output of your new code (the candidate function) matches the output of your old code (the control function). Within the monolith, you can subclass our `ZegoExperiment` class to get automatically get some result metrics (`match`es and `mismatch`es) published to Datadog.
+
```
Comment (dwyery) [inline languages/Python/refactoring.md:10]: > Within the monolith, you can subclass our `ZegoExperiment` class to get automatically get some result metrics

Could this be worded better?

File: languages/Python/refactoring.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        ingredients = experiment.conduct()
+
+        return bake(ingredients)
+```
+
+
+### Feature Flags - not just for features
+
+Feature flags allow partial rollout of all kinds of code, not just new features. Refactored code is particularly well-placed for partial rollout as there is minimal user impact expected (so switching a flag on or off for a particular cohort should result in no surprises for the user) and the in most cases side effects such as writes to the database remain the same. We use [LaunchDarkly](https://launchdarkly.com) to manage flags, which ensures that a client will not switch cohorts unexpectedly.
+
```
Comment (dwyery) [inline languages/Python/refactoring.md:53]: > and the in most cases
```suggestion
 and in most cases
```

**Review by dwyery — APPROVED** [review]
Looks good to me, just a couple of comments around wording.

File: languages/Python/refactoring.md (hunk 1 above)
Comment (hazelw) [reply languages/Python/refactoring.md:10]: probably lol, will have a go

File: languages/Python/refactoring.md (hunk 2 above)
Comment (dwyery) [reply languages/Python/refactoring.md]: Sorry, should not have made that a suggestion lol

File: languages/Python/refactoring.md (hunk 1 above)
Comment (hazelw) [reply languages/Python/refactoring.md:10]: Ended up just moving where this was in the text

**Pull request merged**